test/jdk/tools/jlink/plugins/CDSPluginTest.java

```diff
@@ -87,8 +90,24 @@ public static void main(String[] args) throws Throwable {
        if (Platform.isLinux()) {
            System.out.println("---- Test different platforms scenario ----");
            String jlinkPath = JDKToolFinder.getJDKTool("jlink");
-           String[] cmd = {jlinkPath, "--add-modules", "java.base,java.logging",
-                           "-J-Dos.name=windows", "--generate-cds-archive",
+           // copy over the java.base and java.logging module file to a temporary directory
+           // which we then use as a --module-path during jlink image generation. using such a
+           // separate --module-path will force the JLink task to read the ModuleTarget from
+           // the java.base module-info.class to identify the target platform for the image
+           // being generated.
```

Comment:
I'm a bit uncomfortable with change the test for the CDS plugin as part of this change but can live with it.

Reply:
Early on during this PR, we decided and implemented to read the `ModuleTarget` only if the java.base module's file path doesn't match that of the current platform's file path. If we remove that check and instead always read the ModuleTarget of java.base, in the JLinkTask, we won't need any changes to this test, since the JLinkTask will right find the correct target platform from the module-info.class. Furthermore, rest of the cross platform image generation would work fine too.

Do you think we should remove that file path check? I am guessing the reason why the file path checks were suggested was to reduce any chances of regression with these current changes?

Reply:
My preference would be to drop this as it insane to run with -Dos.name=window set on Linux.

Reply:
I'm working on an update to jlink to use jdk.internal.misc.OperatingSystem instead of os.name (PR #12931), setting os.name on the command line will have no effect. (The test will be updated or removed).